packages/react-i18n/src/babel-plugin.ts

```diff
@@ -0,0 +1,145 @@
+import {TemplateBuilder} from '@babel/template';
+import * as Types from '@babel/types';
+import {NodePath} from '@babel/traverse';
+import stringHash from 'string-hash';
+import path from 'path';
+import fs from 'fs';
+
+interface State {
+  lastImport: NodePath<Types.ImportDeclaration>;
+}
+
+export default function injectWithI18nArguments({
+  types: t,
+  template,
+}: {
+  types: typeof Types;
+  template: TemplateBuilder<Types.ImportDeclaration | Types.ObjectExpression>;
+}) {
+  function fallbackTranslationsImport({id}) {
+    return template(`import ${id} from './translations/en.json';`, {
+      sourceType: 'module',
+    })();
+  }
+  function i18nParams({id, fallbackID, decoratorName}) {
+    return template(
+      `${decoratorName}({
+      id: '${id}',
+      fallback: ${fallbackID},
+      async translations(locale) {
+          try {
+            const dictionary = await import(/* webpackChunkName: '${id}-i18n' */ \`./translations/$\{locale}.json\`);
+            return dictionary;
+          } catch (err) {}
+        },
+    })`,
+      {
+        sourceType: 'module',
+        plugins: ['dynamicImport'],
+        preserveComments: true,
+      },
+    )();
+  }
+  function addI18nArguments({
+    binding,
+    bindingName,
+    filename,
+    filenameRelative,
+    lastImport,
+  }) {
+    if (!hasTranslations(filename)) {
+      return;
+    }
+
+    let fallbackID;
+
+    for (const refPath of binding.referencePaths) {
+      const callExpression: NodePath = refPath.parentPath;
+
+      if (!callExpression.isCallExpression()) {
+        continue;
+      }
+
+      const args = callExpression.get('arguments');
+      if (args.length !== 0) {
+        continue;
+      }
+
+      if (fallbackID == null) {
+        fallbackID = refPath.scope.generateUidIdentifier('en').name;
+        lastImport.insertAfter(fallbackTranslationsImport({id: fallbackID}));
```

Comment:
This also probably fails if there are multiple `i18n` calls?

Reply:
I think it's OK to say it only supports one, we should just make the code reflect that